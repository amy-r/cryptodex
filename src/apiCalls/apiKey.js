const apiKey = 'oU0941gWwanvmtqn7PJQbSAIm7JYqGFP';
export default apiKey