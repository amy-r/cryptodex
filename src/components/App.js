import React, { Component } from 'react';
import './App.css';
import { Route, NavLink, Link } from 'react-router-dom'
import Navigation from './Navigation';
import LandingPage from './Landing';
import SignUpPage from './SignUp';
import SignInPage from './SignIn';
import PasswordForgetPage from './PasswordForget';
import HomePage from './Home';
import AccountPage from './Account';
import * as routes from '../constants/routes';


class App extends Component {
  render() {
    return (
    <div>
      <hr/>
      <Navigation />
      <Route exact path='/' component={LandingPage} />
      <Route exact path='/signup' component={SignUpPage} />
      <Route exact path='/signin' component={SignInPage} />
      <Route exact path='/pw-forget' component={PasswordForgetPage} />
      <Route exact path='/home' component={HomePage} /> 
      <Route exact path='/account' component={AccountPage} /> 
    </div>
    );
  }
}

export default App;
