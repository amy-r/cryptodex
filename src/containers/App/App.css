@import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400,700');

.content-container {
  width: 100%;
  height: 100%;
  position: relative;
  overflow: hidden;
  z-index: -10;
  
  /* Enable GPU Rendering */
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  -o-transform: translateZ(0);
  transform: translateZ(0);
}

.blur {
  /* All the background image supports */
  transform: translate(0px, 0px, 0px);
  background-image: -webkit-linear-gradient(bottom left, rgba(50, 8, 87, 0.8) 0%, rgba(93, 0, 166, 0.8) 45%, rgba(119, 0, 214, 0.6) 60%); 
  background-image: -moz-linear-gradient(bottom left, rgba(50, 8, 87, 0.8) 0%, rgba(93, 0, 166, 0.8) 45%, rgba(119, 0, 214, 0.6) 60%); 
  background-image: -o-linear-gradient(bottom left, rgba(50, 8, 87, 0.8) 0%, rgba(93, 0, 166, 0.8) 45%, rgba(119, 0, 214, 0.6) 60%); 
  background-image: linear-gradient(to top right, rgba(50, 8, 87, 0.8) 0%, rgba(93, 0, 166, 0.8) 45%, rgba(119, 0, 214, 0.6) 60%);
  position: absolute;
  width: 100%;
  height: 100%;
  
  /* blur filters */
  -webkit-filter: blur(20px);
  -o-filter: blur(20px);
  filter: blur(20px);
  
  /* positioning and removing edges */
  top: -14px;
  right: -70px;
  left: -70px;
  padding: 70px;
}

.blur:after {
  content: '';
  position: absolute;
  height: 60%;
  width: 100%;
  z-index: -1;

  /* Setting the gradient. Adds depth, and glass like effect. */
  background-image: -webkit-linear-gradient(289deg, rgba(43,40,50,0.8) 0%, rgba(43,40,50,0.4) 46%, rgba(0,0,0,0.05) 50%);
  background-image: -moz-linear-gradient(289deg, rgba(43,40,50,0.8) 0%, rgba(43,40,50,0.4) 46%, rgba(0,0,0,0.05) 50%);
  background-image: -o-linear-gradient(289deg, rgba(43,40,50,0.8) 0%, rgba(43,40,50,0.4) 46%, rgba(0,0,0,0.05) 50%);
  background-image: linear-gradient(-19deg, rgba(43,40,50,0.8) 0%, rgba(43,40,50,0.4) 46%, rgba(0,0,0,0.05) 50%);

  /* Blur to add depth. High to create smooth transition */
  -webkit-filter: blur(70px);
  -o-filter: blur(70px);
  filter: blur(70px);
}

h1 {
  font-family: 'Open Sans', sans-serif;
}