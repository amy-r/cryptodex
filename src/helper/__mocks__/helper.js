/* eslint-disable */

import {mockDashboard} from '../../helper/mockDashboard';

export const createMarketCap = jest.fn().mockImplementation( () => {
  return [
      { 'currency' : "BTC",
        'close' : "7436.82313",
        'availableSupply' : "16957550.0",
        'marketCap' : 126110300068.1315 }
    ]
})

export const sortByCaps = jest.fn().mockImplementation( () => {
  return mockDashboard
})

export const trimData = jest.fn().mockImplementation( () => {
  return mockDashboard
})


export const createCapPercentage = jest.fn().mockImplementation( () => {
  return mockDashboard
})

export const removeMismatches = jest.fn().mockImplementation( () => {
  return { BTC: 1, ETH: 1}
})

export const findMismatches = jest.fn().mockImplementation( () => {
  return { FAKECOIN: 1 }
})

export const calculateValue = jest.fn().mockImplementation( () => {
  return [7436.82, 387.43,]
})

export const calculatePercent = jest.fn().mockImplementation( () => {
  return [6.67,13.33,20,26.67,33.33]
})