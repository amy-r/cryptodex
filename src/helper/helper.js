export const createMarketCap = (dashboard) => {
  const newDash = dashboard.map( (currency) => {
    currency['marketCap']= parseFloat(currency.close * currency.availableSupply);
    return currency
  });
  return newDash
}

export const sortByCaps = (dashboard) => {
  return dashboard.sort(function(a,b) {
    return b.marketCap - a.marketCap
  })
}

export const trimData = (dashboard) => {
  return dashboard.map( (curr) => {
    return {
      'currency': curr.currency, 
      'marketCap': curr.marketCap
    }
  })
}

export const totalCap = (dashboard) => {
  const total = dashboard.reduce((acc, curr) => {
    return acc + curr.marketCap
  }, 0)
  return total
} 

export const createCapPercentage = (dashboard) => {
  const total = totalCap(dashboard)
  const finalCurrency = dashboard.map( (currency) => {
    const percentage = (currency.marketCap/total) * 100;
    const roundedPercentage = parseFloat(Math.max(percentage).toFixed(2))
    currency['percentage'] = roundedPercentage
    return currency
  })
  return finalCurrency
}