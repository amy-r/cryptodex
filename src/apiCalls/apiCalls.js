import apiKey from './apiKey';

export const fetchApi = async () => {
  try {
    const response = await fetch(`https://api.nomics.com/v1/dashboard?key=${apiKey}`)
    const currData = await response.json()
    console.log(currData)
  } catch (error) {
    throw new Error("error getting data")
  }
}
