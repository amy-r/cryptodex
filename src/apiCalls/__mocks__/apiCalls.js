/* eslint-disable */

import { mockDashboard } from '../../helper/mockDashboard';

export const fetchApi = jest.fn().mockImplementation( () => {
  return Promise.resolve(mockDashboard);
});