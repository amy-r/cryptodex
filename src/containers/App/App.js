import React, { Component } from 'react';
import './App.css';
import { Route, withRouter } from 'react-router-dom';
import LandingPage from '../../components/Landing/Landing';
import HomePage from '../Home/Home';
import { fetchApi } from '../../apiCalls/apiCalls.js';
import { connect } from 'react-redux';
import { addDashboard } from '../../actions/addDashboard';
import * as helper from '../../helper/helper';
import PropTypes from 'prop-types';

export class App extends Component {

  componentDidMount = async () => {
    const dashboard = await fetchApi();
    const dashWithCap = helper.createMarketCap(dashboard);
    const sorted = helper.sortByCaps(dashWithCap);
    const topTenCurrencies = sorted.slice(0, 10);
    const trimmed = helper.trimData(topTenCurrencies);
    const targetData = helper.createCapPercentage(trimmed);
    this.props.addDashboard(targetData);
  }

  render() {
    return (
      <div>
        <Route exact path='/' component={LandingPage} />
        <Route exact path='/home' component={HomePage} /> 
        <div className='content-container'/>
        <div className='blur' />
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addDashboard: (dashboard) => dispatch(addDashboard(dashboard))
});

App.propTypes = {
  addDashboard: PropTypes.func
};

export default withRouter(connect(null, mapDispatchToProps)(App));
