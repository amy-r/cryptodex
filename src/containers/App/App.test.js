/* eslint-disable */
import React from 'react';
import { App, mapDispatchToProps } from './App.js';
import { shallow } from 'enzyme';
import { fetchApi } from '../../apiCalls/apiCalls.js';
import * as helper from '../../helper/helper';
jest.mock('../../apiCalls/apiCalls');
jest.mock('../../helper/helper');

describe('App', () => {

  it('renders without crashing', () => {
    const wrapper = shallow(<App />, {disableLifecycleMethods: true});
    expect(wrapper).toMatchSnapshot();
  });

  it('calls fetchApi on componentDidMount', async () => {
    const wrapper = shallow(<App addDashboard={jest.fn()}/>, {disableLifecycleMethods: false});
    expect(fetchApi).toHaveBeenCalled();
  })

  it('calls createMarketCap on componentDidMount', async () => {
    const wrapper = shallow(<App addDashboard={jest.fn()}/>, {disableLifecycleMethods: false});
    expect(helper.createMarketCap).toHaveBeenCalled();
  })

  it('calls sortByCaps on componentDidMount', async () => {
    const wrapper = shallow(<App addDashboard={jest.fn()}/>, {disableLifecycleMethods: false});
    expect(helper.sortByCaps).toHaveBeenCalled();
  })

  it('calls trimData on componentDidMount', async () => {
    const wrapper = shallow(<App addDashboard={jest.fn()}/>, {disableLifecycleMethods: false});
    expect(helper.trimData).toHaveBeenCalled();    
  })

  it('calls createCapPercentage on componentDidMount', async () => {
    const wrapper = shallow(<App addDashboard={jest.fn()}/>, {disableLifecycleMethods: false});
    expect(helper.createCapPercentage).toHaveBeenCalled();    
  })

  it('calls addDashboard on componentDidMount', async () => {
    const wrapper = shallow(<App addDashboard={jest.fn()}/>, {disableLifecycleMethods: true});
    expect(wrapper.props().addDashboard).toHaveBeenCalled;    
  })
})  

describe('MDTP', () => {
  it('should call dispatch on addDashboard', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addDashboard();
    expect(mockDispatch).toHaveBeenCalled;
  })
})