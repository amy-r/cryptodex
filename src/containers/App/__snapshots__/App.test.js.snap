// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Route
          component={[Function]}
          exact={true}
          path="/"
/>,
        <Route
          component={[Function]}
          exact={true}
          path="/home"
/>,
        <div
          className="content-container"
/>,
        <div
          className="blur"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/home",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "content-container",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "blur",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
/>,
          <Route
            component={[Function]}
            exact={true}
            path="/home"
/>,
          <div
            className="content-container"
/>,
          <div
            className="blur"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/home",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "content-container",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "blur",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
